backend/database/locks.go

```diff
@@ -0,0 +1,61 @@
+package database
```

Comment:
I generally like to leave (or add) credit to the original author, with link when I copy a whole function or file.